pkg/util/docker/cgroup.go

```diff
@@ -577,13 +577,11 @@ func parseCgroupPaths(r io.Reader) (string, map[string]string, error) {
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		l := scanner.Text()
-		if containerID == "" {
-			// Check if this process running inside a container.
-			containerID, ok = containerIDFromCgroup(l)
-			if !ok {
-				log.Debugf("could not parse container id from path '%s'", l)
-				return "", nil, nil
```

Comment:
Why not just keep the previous logic but change the return to `continue`, so then we only ever have parse the ID once?

Reply:
If you had an invalid one in the middle it was getting picked up because we skipped this block